packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -227,9 +227,12 @@ export function createOdspNetworkError(
 			break;
 		case 401:
 		case 403:
-			// The server throws 403 status code with innerMostError code as "serviceReadOnly" for cases where the
-			// database on server becomes readonly. The driver retries for such cases with exponential backup logic.
-			if (innerMostErrorCode === OdspServiceReadOnlyErrorCode) {
+			// After a redirect due to a domain change, auth headers are not sent by the browser and can trigger a 403.
+			if (response?.redirected) {
```

Comment:
Should we assert that response?.redirected => 401 or 403? I.e. that we are not missing cases?

Reply:
Does 404 / @error.redirectLocation case also have redirect property on response?

Reply:
I have not seen a 404 redirect error for me to test that this is valid and works. Do you know how one could trigger a 404 redirection error?

Reply:
Well, the case below (for 404) has code that deals with redirects. I do not know much about how redirects work (i.e. what error codes can be returned) - @jatgarg & @marcmasmsft are rigth people to consult here.